controller/test/frame_detector/frame_detector_test.cpp

```diff
@@ -0,0 +1,198 @@
+#include "frame_detector.h"
+#include "proto_traits.h"
+#include "soft_rx_buffer.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include <string>
+#include <vector>
+using namespace std;
+
+constexpr uint32_t LEN = 20;
+
+using State = FrameDetector<SoftRxBuffer<LEN>, LEN>::State;
+using FrameDetectorTest = FrameDetector<SoftRxBuffer<LEN>, LEN>;
+constexpr uint8_t MARK = static_cast<uint8_t>('.');
+
+SoftRxBuffer<LEN> rx_buf('.');
```

Comment:
I think this should be a local variable in each test, otherwise its state is shared between tests and it makes it non-obvious what exactly is being tested, depending on which other tests have run before this one.

Reply:
DONE